crates/service/src/services/network_checker/live_network_checker.rs

```diff
@@ -23,42 +25,60 @@ lazy_static! {
 #[derive(Debug)]
 pub struct LiveNetworkChecker;
 
+#[async_trait]
 impl NetworkCheckerImpl for LiveNetworkChecker {
-    fn ipv4_lookup(&self, domain: &str) -> Result<Vec<DnsRecord>, DnsLookupError> {
-        let ipv4_lookup = RESOLVER.ipv4_lookup(domain)?;
+    async fn ipv4_lookup(&self, domain: &str) -> Result<Vec<DnsRecord>, DnsLookupError> {
+        let domain = domain.to_string();
+        let ipv4_lookup = tokio::task::spawn_blocking(move || RESOLVER.ipv4_lookup(domain))
```

Comment:
We now use `tokio::task::spawn_blocking` in order for Tokio to execute the code properly.